File: APCCCIDE/FindDialog.vb
Public Class FindDialog
	Private Sub SearchButton_Click(sender As Object, e As EventArgs) Handles SearchButton.Click
		Find()
	End Sub

	Private Sub ReplaceButton_Click(sender As Object, e As EventArgs) Handles ReplaceButton.Click
		Replace()
	End Sub

	Private Sub SearchBox_TextChanged(sender As Object, e As EventArgs) Handles SearchBox.Enter
		Find()
	End Sub

	Private Sub ReplaceBox_TextChanged(sender As Object, e As EventArgs) Handles ReplaceBox.Enter
		Replace()
	End Sub



	Private Sub Find()
		MainForm.Find(SearchBox.Text, RegexCheckBox.Checked, CaseCheckBox.Checked)
	End Sub

	Private Sub Replace()
		MainForm.Replace(SearchBox.Text, ReplaceBox.Text, RegexCheckBox.Checked, CaseCheckBox.Checked)
	End Sub
End Class

File: APCCCIDE/FindDialog.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace APCCCIDE{
    public partial class FindDialog : Form{
        public FindDialog(){
            InitializeComponent();
        }

        private void SearchButton_Click(object sender, EventArgs e){
            Find();
        }

        private void ReplaceButton_Click(object sender, EventArgs e){
            Replace();
        }

        private void SearchBox_KeyDown(object sender, KeyEventArgs e){
            if (e.KeyData == Keys.Enter){
                Find();
                e.Handled = true;
            }
        }

        private void ReplaceBox_KeyDown(object sender, KeyEventArgs e){
            if (e.KeyData == Keys.Enter){
                Replace();
                e.Handled = true;
            }
        }



        private void Find(){
            ((MainForm)this.Owner).Find(SearchBox.Text, RegexCheckBox.Checked, CaseCheckBox.Checked);
        }

        private void Replace(){
            ((MainForm)this.Owner).Replace(SearchBox.Text, ReplaceBox.Text, RegexCheckBox.Checked, CaseCheckBox.Checked);
        }
    }
}
